rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1426,32 +1581,69 @@ def batch_predict(
         )
 
         out = {}
-        if self.config[INTENT_CLASSIFICATION]:
-            if self.all_labels_embed is None:
-                _, self.all_labels_embed = self._create_all_labels()
-
-            # get _cls_ vector for intent classification
-            cls = self._last_token(text_transformed, sequence_lengths)
-            cls_embed = self._tf_layers[f"embed.{TEXT}"](cls)
 
-            # pytype cannot infer that 'self._tf_layers[f"loss.{LABEL}"]' has methods
-            # like '.sim' or '.confidence_from_sim'
-            # pytype: disable=attribute-error
-            sim_all = self._tf_layers[f"loss.{LABEL}"].sim(
-                cls_embed[:, tf.newaxis, :], self.all_labels_embed[tf.newaxis, :, :]
-            )
-            scores = self._tf_layers[f"loss.{LABEL}"].confidence_from_sim(
-                sim_all, self.config[SIMILARITY_TYPE]
-            )
-            # pytype: enable=attribute-error
-            out["i_scores"] = scores
+        if self.config[INTENT_CLASSIFICATION]:
+            self._batch_predict_intents(out, sequence_lengths, text_transformed)
 
         if self.config[ENTITY_RECOGNITION]:
-            logits = self._tf_layers["embed.logits"](text_transformed)
-            pred_ids = self._tf_layers["crf"](logits, sequence_lengths - 1)
-            out["e_ids"] = pred_ids
+            self._batch_predict_entities(out, sequence_lengths, text_transformed)
 
         return out
 
+    def _batch_predict_entities(
+        self,
+        out: Dict[Text, Any],
+        sequence_lengths: tf.Tensor,
+        text_transformed: tf.Tensor,
+    ) -> None:
+        prev_logits = None
+
+        for crf_layers in self._crf_layers:
```

Comment:
in train method, this `for` is outside entity helper, let's unify this. Either outside in both train and predict helpers or inside in both